rapids-cmake/cython-core/create_modules.cmake

```diff
@@ -0,0 +1,158 @@
+# =============================================================================
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+rapids_cython_create_modules
+----------------------------
+
+.. versionadded:: v24.02.00
+
+Generate C(++) from Cython and create Python modules.
+
+.. code-block:: cmake
+
+  rapids_cython_create_modules([CXX] [SOURCE_FILES <src1> <src2> ...] [LINKED_LIBRARIES <lib1> <lib2> ... ]  [INSTALL_DIR <install_path>] [MODULE_PREFIX <module_prefix>] )
+
+Creates a Cython target for each provided source file, then adds a
+corresponding Python extension module. Each built library has its RPATH set to
+$ORIGIN.
+
+.. note::
+  Requires :cmake:command:`rapids_cython_init` to be called before usage.
+
+``CXX``
```

Comment:
Is there a reason why we'd ever want this to be false? I'd just force this to true and remove the parameter.

Reply:
When I originally wrote rapids-cython, I wanted to allow for C projects in case we had something like ucx-py or dask-cuda writing C (not C++) extensions. In practice, that hasn't turned out to be the case AFAIK. That being said, I'd like to leave this in place for now because I've endeavored to make the new functions drop-in replacements for the old ones and this would be an API break. I plan future changes that will be API breaking, so I can make this change then. For now, the scikit-build-core PRs in other repos are made much cleaner by avoiding to have to change any CMake other than the `include(rapids-cython)`.